### PR TITLE
makefiles/suit: increase SOCK_URLPATH_MAXLEN

### LINKED ISSUE #9: suit: url are to long for net_sock_util
#### Description

When playing around with the extension and naming of the manifest I found that when extending the url by two bytes I got the following error:

```
2019-05-20 16:48:19,061 - INFO # suit_coap: trigger received
2019-05-20 16:48:19,072 - INFO # suit_coap: downloading "coap://[fd00:dead:beef::1]/fw/samr21-xpro/tests_suit_update-riot.suitv4_signed.latest-1.bin"
2019-05-20 16:48:19,073 - INFO # urlpath: �Durlsplit
2019-05-20 16:48:19,076 - INFO # suit_coap: error getting manifest
```

Looking into it `len("fw/samr21-xpro/tests_suit_update-riot.suitv4_signed.latest-1.bin") = 65` and 
https://github.com/future-proof-iot/RIOT/blob/59b4dcffed6b8f32e6fba01e262e941d7413f7ca/sys/include/net/sock/util.h#L107-L108  

#### Steps to reproduce the issue

Change the signed manifest extension. Publish new firmware

#### Proposed Fixes

* Simply define a bigger value
* When calling `make` define a `CFLAGS` for `SOCK_URLPATH_MAXLEN` that is calculated from the size of the URL extension (with some kind of maximum value).

What do you think?

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR makes `sys/net/sock/util.h` variables configurable and with this we can redefine SOCK_URLPATH_MAXLEN at compilation time to increase it when the naming for the URL file extension is too large.

### Testing procedure

in `makefiles/suit.v4.inc.mk` modify naming of `SUIT_MANIFEST_SIGNED_LATEST ?= $(BINDIR_APP)-riot.suitv4_signed.latest.bin`.

Follow [README](https://github.com/future-proof-iot/RIOT/blob/suit-dev/examples/suit_update/README.md)

### Issues/PRs references

Fixes #9

### CODE REVIEW

**Comment by fjmolinas** [comment]
@kaspar030 didn't realize you had opened https://github.com/RIOT-OS/RIOT/pull/11558. I'll rebase on top of that.

File: makefiles/suit.v4.inc.mk [hunk 1]
```diff
@@ -62,6 +71,9 @@ suit/notify: | $(filter suit/publish, $(MAKECMDGOALS))
 		--payload "$(SUIT_COAP_ROOT)/$$(basename $(SUIT_MANIFEST_SIGNED_LATEST))" && \
 		echo "Triggered $(SUIT_CLIENT) to update."
 
+suit/delkeys:
+	rm -f *key*
```
Comment (kaspar030) [inline makefiles/suit.v4.inc.mk]: this is unrelated and also strikes me as a bit extreme. ;)

File: sys/include/net/sock/util.h [hunk 2]
```diff
@@ -99,13 +99,48 @@ bool sock_udp_ep_equal(const sock_udp_ep_t *a, const sock_udp_ep_t *b);
  * @name helper definitions
  * @{
  */
+<<<<<<< HEAD
```
Comment (kaspar030) [inline sys/include/net/sock/util.h]: rebase mishap

**Review by kaspar030 — CHANGES_REQUESTED** [review]

File: sys/include/net/sock/util.h (hunk 2 above)
Comment (fjmolinas) [reply sys/include/net/sock/util.h]: ups

File: makefiles/suit.v4.inc.mk (hunk 1 above)
Comment (fjmolinas) [reply makefiles/suit.v4.inc.mk]: Yes should be in different PR, sorry. But would removing the specific file names be better then?

Comment (kaspar030) [reply makefiles/suit.v4.inc.mk]: :+1:

imagine the surprise when someone starts working on e.g., "hash_key.c" and runs "make delkeys" to get to a clean state before committing...

File: makefiles/suit.v4.inc.mk [hunk 3]
```diff
@@ -10,6 +10,15 @@ SUIT_MANIFEST_LATEST ?= $(BINDIR_APP)-riot.suitv4.latest.bin
 SUIT_MANIFEST_SIGNED ?= $(BINDIR_APP)-riot.suitv4_signed.$(APP_VER).bin
 SUIT_MANIFEST_SIGNED_LATEST ?= $(BINDIR_APP)-riot.suitv4_signed.latest.bin
 
+SUIT_URLPATH ?= "$(SUIT_COAP_BASEPATH)/$(shell echo $$(basename $(SUIT_MANIFEST_SIGNED_LATEST)))"
+SUIT_URLPATH_LEN ?= $(shell expr length \"$(SUIT_URLPATH)\")
+SOCK_URLPATH_MAXLEN ?= $(shell echo "$$((($(SUIT_URLPATH_LEN) / 32 + 1 ) * 32))")
+ifeq ($(shell expr $(SOCK_URLPATH_MAXLEN) \>= 128), 1)
+   export CFLAGS += -DSOCK_URLPATH_MAXLEN=128
```
Comment (kaspar030) [inline makefiles/suit.v4.inc.mk]: IMO we can just use this...

Comment (fjmolinas) [reply makefiles/suit.v4.inc.mk]: Would you be ok with an error message if its too long then?

Comment (fjmolinas) [reply makefiles/suit.v4.inc.mk]: The function states that when failing. So scratch that.
